CONTRIBUTING.md

```diff
@@ -23,3 +23,55 @@ use GitHub pull requests for this purpose. Consult [GitHub Help] for more
 information on using pull requests.
 
 [GitHub Help]: https://help.github.com/articles/about-pull-requests/
+
+## Building Dagger
+
+Dagger is built with [`bazel`](https://bazel.build).
+
+### Setting up Bazel
+
+* Go to [http://bazel.build](http://bazel.build), click on `GET BAZEL`, and install Bazel.
+* In *Android Studio*, go to `Preferences > Plugins`.
+  * search for `bazel` and install the plugin.
+  * If no result shows up, click on `Search in repositories`, search for `bazel` and install the plugin.
+  * If you still get no result, then the bazel plugin isn't available for your version of Android Studio.
+    * Go to [Integrating Bazel with IDEs](https://docs.bazel.build/versions/master/ide.html), click on `Android Studio plugin`.
+	* The `COMPATIBILITY` column indicates which versions of Android Studio have an available Bazel plugin.
+	* You can find the corresponding numbering scheme in the file names of [Android Studio downloads](https://developer.android.com/studio/#downloads). You may need to [download a Preview](https://developer.android.com/studio/preview/).
+
```

Comment:
I think a reference to the bazel IJ plugin would suffice here vs full installation steps.

Reply:
Unfortunately not. When we looked at this at droidcon, the plugin as listed when you search plugins is actually shown as incompatible with all the versions of IJ and AS we tried :|

Reply:
Isn't that an issue you should file on that project then? Is every Bazel project supposed to detail the steps for installing the IDE plugin?

Reply:
Yes, but that could be done in parallel right?

Reply:
Also - it's filed. No support for AS 3.2 nor IntelliJ 2018.2

Maybe the wording should be tweaked that this is only necessary for 2018.2+ or AS 3.2+?

Reply:
Projects don't exist in a vacuum. A number of people who could contribute have felt friction, and didn't know how to move forward. To get people to build and contribute to the Dagger project as fast as possible, we did a run through and we captured every issue that we went through and wrote that down.

Reply:
Sure but if your left shift key is broken you don't write down to use the right shift key in these notes. This whole section should be on the bazel plugin readme, not here.

Reply:
If 50% people have a broken left shift key, you might still want to write it down :) 

That being said, yes, ideally this would all be in the bazel plugin doc. Or, even better, we wouldn't have this issue at all because the plugin would exist for all IJ builds...

Honestly we could also just jump to specifics and say "you have to install Android Studio 3.2+" and then remove this section.

Reply:
If they own a new MacBook Pro that number isn't far off...

My concern here is the long-term maintenance of this section. For better or worse, Dagger's documentation is updated infrequently. Having something that ages better would be preferable to me. So something along those lines sounds fine because while it's annoying to require such a new version now, in a year these docs will still be accurate and no one will think twice about needing AS 3.3 or newer.